src/util/mod.rs

```diff
@@ -49,20 +48,29 @@ pub mod collections;
 #[cfg(target_os="linux")]
 mod thread_metrics;
 
-pub fn limit_size<T: Message + Clone>(entries: &mut Vec<T>, max: u64) {
-    if entries.is_empty() {
-        return;
-    }
-
-    let mut size = Message::compute_size(&entries[0]) as u64;
-    let mut limit = 1usize;
-    while limit < entries.len() {
-        size += Message::compute_size(&entries[limit]) as u64;
+pub use fnv::{FnvHashMap as HashMap, FnvHashSet as HashSet, FnvBuildHasher as BuildHasherDefault};
+
+pub fn get_limit_at_size<'a, T: Message + Clone, I: IntoIterator<Item = &'a T>>(entries: I,
+                                                                                max: u64)
+                                                                                -> usize {
+    let mut iter = entries.into_iter();
```

Comment:
if max is NO_LIMIT, I think we can return the limit directly, no need to compute the sizes. 

And in our raft, most of the places use NO_LIMIT.